src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,286 @@
+
+        for (Object sample: sampleList) {
+            Map<String, String> sampleMap = mapper.convertValue(sample, Map.class);
+            if (!igoCmoRequestFilter && !isCmoRequest) {
+                if (isValidNonCmoSample(sampleMap)) {
+                    validSampleList.add(sample);
+                }
+            } else {
+                if (isValidCmoSample(sampleMap, isCmoRequest, hasRequestId)) {
+                    validSampleList.add(sample);
+                }
+            }
+
+        }
+
+        if (validSampleList.size() > 0) {
+            if (validSampleList.size() < numOfSamples) {
+                String newSamplesString = mapper.writeValueAsString(validSampleList.toArray(new Object[0]));
+                requestJsonMap.replace("samples", newSamplesString);
+                LOG.info("CMO request passed sanity checking - missing samples...");
+                requestStatusLogger.logRequestStatus(requestJson,
+                        RequestStatusLogger.StatusType.CMO_REQUEST_MISSING_REQ_FIELDS);
+            }
+            return mapper.writeValueAsString(requestJsonMap);
+        }
+        LOG.error("Request failed sanity checking - logging request status...");
+        requestStatusLogger.logRequestStatus(requestJson,
+                RequestStatusLogger.StatusType.CMO_REQUEST_FAILED_SANITY_CHECK);
+        return null;
+    }
+
+    /**
+     * isValidCmoSample receives a sampleMap
+     * 
+     * <p><ul>
+     * <li>Checks if the sample has all the required fields
+     * <li>If RequestId, InvestigatorSampleId, CmoPatientId,
+     * SpecimenType, SampleType or NormalizedPatientId are missing, returns false
+     * </ul>
+     */
+    private boolean isValidCmoSample(Map<String, String> sampleMap,
+            boolean isCmoRequest, boolean hasRequestId) {
+        if (sampleMap.isEmpty() || sampleMap == null) {
+            return false;
+        }
+        if (isCmoRequest && !hasBaitSet(sampleMap)) {
+            return false;
+        }
+        if (!hasRequestId || !hasInvestigatorSampleId(sampleMap)
+                || !hasCmoPatientId(sampleMap) || !hasSpecimenType(sampleMap)
+                || !hasSampleType(sampleMap) || !hasNormalizedPatientId(sampleMap)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * isValidNonCmoSample receives a sampleMap
+     * 
+     * <p><ul>
+     * <li>Check if the sample has all the required fields
+     * <li>If baitSet or cmoPatientId is missing, returns false
+     * </ul>
+     */
+    private boolean isValidNonCmoSample(Map<String, String> sampleMap) {
+        if (sampleMap.isEmpty() || sampleMap == null) {
+            return false;
+        }
+        if (!hasBaitSet(sampleMap) || !hasNormalizedPatientId(sampleMap)) {
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isCmoRequest(Map<String, Object> requestJsonMap) {
+        if (Strings.isBlank(requestJsonMap.get("cmoRequest").toString())) {
```

Comment:
need to check what the value stored in `cmoRequest` actually is
```
return Boolean.valueOf(requestJsonMap.get("cmoRequest").toString())
```

Reply:
another thing I just noticed also... if you try to invoke `.toString()` on a null object then a null pointer exception would be thrown. 

I think `toString()` should be removed here.